src/integrationtests/resources/com/aws/greengrass/integrationtests/deviceauth/config.yaml

```diff
@@ -0,0 +1,72 @@
+---
+services:
+  aws.greengrass.Nucleus:
+    configuration:
+      runWithDefault:
+        posixUser: nobody
+        windowsUser: integ-tester
+      logging:
+        level: "DEBUG"
+  aws.greengrass.clientdevices.Auth:
+    configuration:
+      deviceGroups:
+        formatVersion: "2021-03-05"
+        definitions:
+          myThing:
+            selectionRule: "thingName:myThing"
+            policyName: "thingAccessPolicy"
+        policies:
+          thingAccessPolicy:
+            policyStatement1:
+              statementDescription: "mqtt connect"
+              effect: ALLOW
+              operations:
+                - "mqtt:connect"
+              resources:
+                - "mqtt:${iot:Connection.Thing.ThingName}:foo"
```

Comment:
i don't think we should allow substitution here, just in the third part of the resource

Reply:
Isn't this the primary use case @jcosentino11?

Reply:
While I agree this doesn't make practical sense (since it can only be filled with `mqtt:topic`, `mqtt:clientId`), is it actually reasonable for us to prevent it? Or would preventing it actually be more difficult

Reply:
> since it can only be filled with

i don't think we enforce this though?  the above policy right now is considered valid and auth passes, (`mqtt:myThing:foo` in this case)

Reply:
It can only be useful when filled with those values.

Reply:
gotcha. yeah agreed that it'll only complicate things, fine w/ leaving as is

Reply:
Glad to see this feature being implemented!

+1 to Joe's initial thought - I'd suggest validating config when it's loaded and explicitly forbidding this (then you don't pay a penalty in the authZ path). Should be trivial, right? It's just a prefix match against the supported resource types.

> i don't think we enforce this though? the above policy right now is considered valid and auth passes, (`mqtt:myThing:foo` in this case)

Joe - are you saying that this still gets interpreted as a client ID resource type and allows a connection? I hope not!

Reply:
yep what michael said, hi @jbutler :)